tools/@aws-cdk/prlint/lint.ts

```diff
@@ -272,11 +304,92 @@ export class PullRequestLinter {
     }
   }
 
+  /**
+   * Whether or not the codebuild job for the given commit is successful
+   *
+   * @param sha the commit sha to evaluate
+   */
+  private async codeBuildJobSucceeded(sha: string): Promise<boolean> {
+    const statuses = await this.client.rest.repos.listCommitStatusesForRef({
+      owner: this.prParams.owner,
+      repo: this.prParams.repo,
+      ref: sha,
+    });
+    return statuses.data.some(status => status.context === CODE_BUILD_CONTEXT && status.state === 'success');
+  }
+
+  public async validateStatusEvent(pr: GitHubPr, status: StatusEvent): Promise<void> {
+    if (status.context === CODE_BUILD_CONTEXT && status.state === 'success') {
+      await this.assessNeedsReview(pr);
+    }
+  }
+
+  /**
+   * Assess whether or not a PR is ready for review from a core team member.
+   * This is needed because some things that we need to evaluate are not filterable on
+   * the builtin issue search. A PR is ready for review when:
+   *
+   *   1. Not a draft
+   *   2. Does not have any merge conflicts
```

Comment:
What happens when merge conflicts arise _after_ this has run? Since a merge conflict comes from a different PR getting merged, I don't think this test would get triggered again. 

I think I'm okay with just saying that some merge conflicts might fall through the cracks (and at least at some point it was clean). But want to make sure thats an explicit decision

Reply:
Good point, let me see if there is a way to handle this.

Reply:
I think it might be best to handle this as part of the stale issue workflow. It doesn't look like there is an event that triggers when a PR gets merge conflicts. When the stale issue workflow runs to commit on the issue we can also have it remove the `pr/needs-review` label if it exists. There may be some lag there, but I think it is the best we can do.